### ISSUE TITLE
local port is in "close_wait" state

### ISSUE THREAD

**Issue by fenjuly** [shadowsocks/shadowsocks-android#1410]
### Environment

* Android version: 7.1.1
* Device: Samsung Galaxy Note8
* Shadowsocks version: v4.2.5
* Last version that did not exhibit the issue (if applicable):

### What did you do?
Just start the shaodwsocks by using any port, like the default port '1080'.

### What did you expect to see?
shadowsocks works normally

### What did you see instead?
I started the shadowsocks, after a while, it was found that the whole network of my phone was broken, there is no network connection. but, when I closed the shadowsocks, and change the local port to another number, and start shadowsocks agin, it works fine.
so I connect my phone to pc, search the previous port in adb shell, I found that the port was in "close_wait" state.
![image](https://user-images.githubusercontent.com/4019015/31572571-1580599a-b0dc-11e7-8fdc-9a79a0ced84a.png)

any idea of that?

**Comment by madeye** [comment]
CLOSE_WAIT is a normal status.

I don't think your problem is related to local port. Make sure the background service of shadowsocks is not killed by your system, which is a common issue for Samsung devices.

**Comment by bronco** [comment]
Do you install "Brevent(黑域)"? I've same issue here, also in Samsung note 8. 
After the network lost connection, i have to force stop the ss, simple disconnect and re-connect in ss is useless. But if i change the local port in ss config and connect, without force stop the ss, network comes back to normal.
Now i uninstall brevent, and it seems that the ss work fine, but more observation is needed. So if you install brevent too, maybe you could try uninstall it.

**Comment by 925001777** [comment]
我就想问各位大侠敢不敢不说英文，直接中文OK? :）

**Comment by fenjuly** [comment]
@bronco No, I havn't installed Brevent. Is there still no trouble with your ss after uninstalling the Brevent?

**Comment by bronco** [comment]
@fenjuly Until now, seems everything works as expected. But even with Brevent installed, the network may works well for few days, so i could not confirm that the network issue is caused by Brevent.

Anyway, if you didn't installed Brevent, maybe it is your misconfiguration.
There are several items of setting you should config:
1. Smart Manager-->Battery-->Unmonitored apps;
2. Smart Manager-->App management-->Manage Autorun;

Besides,  if you add ss into Security Folder/MyKnox, you need add the ss-secured(the ss in Security Folder) into white list as well.

**Comment by fenjuly** [comment]
@bronco I checked my battery setting, and found that ss was monitored.
now, I add ss into unmonitored list. I will continue to observe the performance.
Thanks for your advice.

**Comment by HankAviator** [comment]
might be relevant #1414

**Comment by fenjuly** [comment]
@bronco Even I added ss into battert unmonitored list, the phone still lost network connection.

**Comment by bronco** [comment]
@fenjuly Well, after i uninstalled the brevent, the ss works just fine. So maybe you could using logcat to get some log? Or uninstall and reinstall the ss?
Besides, Have you update the system to the latest version? I'm using QI9 now.

**Comment by SoDa-GitHub** [comment]
Any solution? My Note8 can't use shadowsocks too. Shadowsocks will work normally for only several minutes after restarting phone each time. Then the network will broken. I have added ss to battery unmointored list and autorun list but there is no effect. In addition, my computer works well.

**Comment by SoDa-GitHub** [comment]
It should be a bug. I have used ssr for one week. It works perfectly. But ss can't work normally.

**Comment by madeye** [comment]
@fenjuly I think you may submit this issue to Samsung. It looks they messed up the local sockets.

If possible, try any AOSP ROMs instead of Samsung's, since we never saw this kind of issue on an AOSP system.

**Comment by bronco** [comment]
@fenjuly So sad that this problem is still exists in my note8😭, although it only appears in some specific conditions, which i have no clue at all. Force stop and re-connect can fix the problem.

In my former S7E, everything works fine. So this issue may relates to note8's ROM or the new Smart Manager. It's said that the new Android OREO will be published in early 2018 by Samsung, so good luck to all of us.

**Comment by Mugenxxx** [comment]
So sad not to see the precise solution.

**Comment by madeye** [comment]
Please try this APK: https://www.dropbox.com/s/yalz5oo0sdtp7v8/shadowsocks-nightly-4.4.0.apk

**Comment by fenjuly** [comment]
@madeye I have tried this apk, but the bug was not fixed.

**Comment by bronco** [comment]
Seems the new version(4.4.x) works great in my note8.

**Comment by SoDa-GitHub** [comment]
I think samsung may has fixed the bug in last OTA update.  Now, SSR seems to work great, too. (I didn't update SSR) I need more time to check it.

**Comment by SoDa-GitHub** [comment]
I must withdraw my words. It seems to be still instability now. But really better than before.

**Comment by AltanW** [comment]
I've met this problem with my note8 as well, it bothers me a lot and I'm going crazy……

**Comment by lifeva** [comment]
me too

**Comment by lifeva** [comment]
4.4.6still have this problem

**Comment by evenardo** [comment]
I also have this problem today , but I have a solution seems precise : 

SETTINGS -> GENERAL MANAGEMENT -> Reset -> Reset network settings 

Take a deep breath , this is very important . Hands together , or be fail ! 

The Free World come back !!!

**Comment by CGDF-Github** [comment]
i renamed the applicationId and package name("com.github.shadowsocks") and tested the new apk for about one week in my Samsung phone. the abnormal phenomena never appeared again even once.
i believe that Samsung's ROM is detecting app's name and doing some malicious activity.
so is MIUI.

**Comment by guangmuzhu** [comment]
I had a similar situation but with a few differences. If I set using the VPN for all connections the SS would work fine, but if not, the app would have the same problem as above.

I found an ez way to fix it. There's a system setting for VPN to make it ALWAYS open. I don't know its English name but it should be found in the VPN settings. The system had no reason to kill an always opening app, I think.

Samsung S8, HK version.
